<launch>
    <arg name="x_pos" default="-2.0"/>
    <arg name="y_pos" default="-0.5"/>
    <arg name="z_pos" default="0.0"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/turtlebot3_world.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find my_car)/urdf/my_car.urdf.xacro" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
            <param name="publish_frequency" type="double" value="50.0"/>
    </node>

    <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher"/>
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model my_car -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />

    <node pkg="my_car" type="my_car_drive" name="my_car_drive" output="screen"/>
    <node pkg="my_car" type="arduino_drive" name="arduino_drive" output="screen"/>

    <node pkg="joy" type="joy_node" name="joy_node">
        <param name="dev" type="string" value="/dev/input/js0"/>
        <param name="deadzone" value="0.12"/>
    </node>
</launch>
